pykokkos/lib/ufuncs.py

```diff
@@ -260,13 +268,18 @@ def sign(view):
 
 @pk.workunit
 def add_impl_1d_double(tid: int, viewA: pk.View1D[pk.double], viewB: pk.View1D[pk.double], out: pk.View1D[pk.double], ):
-    out[tid] = viewA[tid] + viewB[tid]
+    out[tid] = viewA[tid] + viewB[tid % viewB.extent(0)]
 
 
 @pk.workunit
 def add_impl_1d_float(tid: int, viewA: pk.View1D[pk.float], viewB: pk.View1D[pk.float], out: pk.View1D[pk.float]):
     out[tid] = viewA[tid] + viewB[tid]
 
+@pk.workunit
+def add_num_2d_impl_1d_double(tid: int, viewA: pk.View2D[pk.double], viewB: pk.View1D[pk.double], out: pk.View2D[pk.double]):
+    for i in range(viewA.extent(1)):
+        out[tid][i] = viewA[tid][i] + viewB[i % viewB.extent(0)]
+
```

Comment:
Why does the workunit start with `add_num` instead of just `add`? Also same question for the modulo

Reply:
Renamed it to `add_2d...`. Used modulo for the same reason here to prevent duplication of workunits for different combinations of inputs.